connectors.go

```diff
@@ -62,10 +63,19 @@ type ObjectMetadataConnector interface {
 	ListObjectMetadata(ctx context.Context, objectNames []string) (*ListObjectMetadataResult, error)
 }
 
+var ErrPostAuthDataIsNotSet = errors.New("post authentication metadata is not set")
+
+// AuthMetadataConnector is an interface that extends the Connector interface with
+// the ability to retrieve metadata information about authentication.
 type AuthMetadataConnector interface {
 	Connector
 
+	// GetPostAuthInfo returns authentication metadata.
 	GetPostAuthInfo(ctx context.Context) (*common.PostAuthInfo, error)
+
+	// SetPostAuthInfo sets, configures connector to use authentication metadata acquired from GetPostAuthInfo.
+	// This data is required for connector to function properly and completes setup.
+	SetPostAuthInfo(ctx context.Context, info *common.PostAuthInfo) error
```

Comment:
This is the **core change** in this PR that triggered implementation of this method in Docusign.